### ISSUE TITLE
pyGSTi fails to generate plots without X

### ISSUE THREAD

**Issue by marcusps** [pyGSTio/pyGSTi#1]
See error messages in the attached (as PNG and zipped Jupyter notebook). This happens in the `00 Quick and easy GST.ipynb` file included with pyGSTi.

 It seems to indicate that using the default configuration, matplotlib will try to use the QtAgg backend to plot, but then it can't plot because there is no DISPLAY variable set (X is not running).

This error can be reproduced by using a docker image built from [BBN-Q/pygsti-docker](https://github.com/BBN-Q/pygsti-docker) (so it runs on Ubuntu 14.04, and all python library dependencies are explicitly listed there).

![display-error](https://cloud.githubusercontent.com/assets/232049/13324182/91a40e58-dbaa-11e5-84be-49c094d084eb.png)

[00 Quick and easy GST MPS.ipynb.zip](https://github.com/pyGSTio/pyGSTi/files/146805/00.Quick.and.easy.GST.MPS.ipynb.zip)

**Comment by marcusps** [comment]
A simple work around is to add 

```python
import matplotlib
matplotlib.use('Agg')
```

at the top of the notebook and restart the kernel.

I am not sure how this interacts with inline plots, though.

**Comment by enielse** [comment]
Just to follow up, I think we've addressed this issue in a somewhat roundabout way in that as of pyGSTi version 0.9.4 we don't use matplotlib (nor X) to generate report plots by default.  There is still, however the option of generating PDF-format reports which still use matplotlib, and if one does this in a notebook the workaround suggested by Marcus above is the best and only thing I know to skirt the issue of Matplotlib's "inline" backend" not being able to write PDFs.

So in summary:
- with pyGSTi 0.9.4 and later you probably won't run into a `RuntimeError: Invalid DISPLAY variable` issue.
- but if you do,  adding 
```
import matplotlib
matplotlib.use('Agg')
```
 to the beginning of your notebook should fix the problem.
